backend/common/src/main/java/ai/verta/modeldb/common/futures/FutureExecutor.java

```diff
@@ -37,8 +58,24 @@ public static FutureExecutor newSingleThreadExecutor() {
     return makeCompatibleExecutor(Executors.newSingleThreadExecutor());
   }
 
+  private Runnable wrapWithTimer(Runnable r) {
+    return () -> {
```

Comment:
Is it possible to make this return `r` unchanged if the OpenTelemetry instance hasn't been configured

Reply:
There isn't a way to know if OpenTelemetry has been configured or not... so unfortunately not.

Reply:
Another option (that I will probably end up with) is to inject the OpenTelemetry MeterProvider into this class, rather than rely on the global. Right now...still experimenting, though!